pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -55,10 +56,21 @@ func discoverKubeAPIServerEndpoint() (string, error) {
 	if err != nil {
 		return "", errors.Wrap(err, "unable to get config to talk to kubernetes api server")
 	}
-	if len(cfg.Host) == 0 {
+
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get client from the given config")
+	}
+
+	host, err := client.ConfigV1().Infrastructures().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get cluster infrastructure resource")
+	}
+	// get API serverl internal url of format https://api-int.abc.devcluster.openshift.com:6443
+	if len(host.Status.APIServerInternalURL) == 0 {
```

Comment:
I prefer `host.Status.APIServerInternalURL == ""` unless we want to standardize on using `len` everywhere.